### ISSUE TITLE
Not working with different transports

### ISSUE THREAD

**Issue by jorgecuesta** [senecajs/seneca-mesh#63]
Hi, I try to use it with different transports like amqp, redis, tcp and not get working properly.
- TPC: It send message, but seneca-transport has declared non close issue with tcp. After a restart of node it fail.
- AMQP/REDIS: With both of this same issue, no matching pattern bla bla...

To check this I use your examples on 30-multicast-discovery.

I'm really new on this of seneca and maybe lose something.

Can you check this? I'm working on a [fork](https://github.com/jorgecuesta/seneca-mesh) to help you with examples because are not working as is.

Regards.

**Comment by danilotorrisi** [comment]
I have the same issue here. I'm trying use amqp transport with seneca-mesh without any luck. 

I've tried to debug the issues and I found out that the cluster member doesn't have a valid entry.

`[
  {
    "pin": "null:true",
    "instance": "tscm6v3n2nm3/1478617979800/54959/3.1.0/-"
  }
]`

From what I understood the `transport_listen` function gets called to initialize the amqp transport. Then it calls the this.prior and via the callback it calls the `join` function passing the out parameter from the callback. The issues is that the out parameter is undefined, and thus the pin is defined by default to 'null:true'.

Is there anyone who actually made it work?

Best,
Danilo
